### PR TITLE
[NN] Add commonly used GNN models from examples to dgl.nn modules.

### PR DESCRIPTION
**Pull request by yzh119**
## Description
For some commonly used GNN models(GAT, GraphSAGE, ChebyNet, GIN, ...), user may find importing layers (nn.Module for PyTorch and nn.Block for MXNet) more convenient (as in #742). Now we already support GraphConv in dgl.nn.*.conv, and @jermainewang finished RGCN; this PR aims to take more models from dgl examples to dgl.nn modules, for usability.

My concern is that should we preserve the code in dgl examples or let them import dgl.nn modules, the second way may break readablity of our examples. Any comments are welcomed.

### Models
- [ ] MXNet support
- [x] GATConv
- [x] SAGEConv
- [x] APPNPConv
- [x] ChebyNet
- [x] GIN
- [x] SGC
- [x] NNConv
- [ ] PointConv
- [ ] SplineConv
- [x] GatedGraphConv
- [x] EdgeConv (#789 )
- [x] DynamicEdgeConv (#789 )
- [x] AGNN
- [x] GMMConv

### Dense Layers
- [x] DenseGraphConv
- [x] DenseSAGEConv
- [ ] DenseChebConv (@BarclayII )

### Functions
- [x] `transform.khop_graph`: Return the graph that includes all k-hop neighbors of original graph as edges.
- [x] `transform.khop_adj`: Return the matrix of A^k where A is the adjacency matrix of given graph.
- [x] `transform.lambda_max`: Maximum eigenvalue of the normalized laplacian matrix of given graph.
- ...

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

## Changes
- TODO

### CODE REVIEW

File: python/dgl/nn/pytorch/conv.py [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
 from ... import function as fn
+from ...batched_graph import broadcast_nodes
+from ...transform import laplacian_lambda_max
+from .softmax import edge_softmax
+
+__all__ = ['GraphConv', 'GATConv', 'TAGConv', 'RelGraphConv', 'SAGEConv',
+           'SGConv', 'APPNPConv', 'GINConv', 'GatedGraphConv', 'GMMConv',
+           'ChebConv', 'AGNNConv', 'NNConv', 'DenseGraphConv', 'DenseSAGEConv']
+
+# pylint: disable=W0235
```
[the remark this answers is not in the document]
Comment (yzh119) [reply python/dgl/nn/pytorch/conv.py]: Yes, it's only for `Identity`.

File: python/dgl/transform.py [hunk 2]
```diff
@@ -169,4 +241,39 @@ def to_bidirected(g, readonly=True):
...
+    if isinstance(g, BatchedDGLGraph):
+        g_arr = unbatch(g)
+    else:
+        g_arr = [g]
+
+    rst = []
+    for g_i in g_arr:
+        n = g_i.number_of_nodes()
+        adj = g_i.adjacency_matrix_scipy(return_edge_ids=False).astype(float)
+        norm = sparse.diags(asnumpy(g_i.in_degrees()).clip(1) ** -0.5, dtype=float)
```
[the remark this answers is not in the document]
Comment (yzh119) [reply python/dgl/transform.py]: We do not plan to handle the case of directed graph.

[the remark this answers is not in the document]
Comment (yzh119) [reply python/dgl/transform.py]: Note that we clip the degrees lower than 1, so there is no need to worry.
